### PR TITLE
WT-12487 Session join API: remove feature (Including join: cursor)

### PR DESCRIPTION
**Pull request by tod-johnson-mongodb**

### CODE REVIEW

**Comment by tod-johnson-mongodb** [comment]
Don't review yet. Putting back to draft. It passed evergreen once. When I created this PR it kicked off evergreen again. Waiting for that to finish before asking for review.

**Comment by tod-johnson-mongodb** [comment]
Ready for review. Evergreen passed. https://spruce.mongodb.com/version/66030e32bd735f0007001f1a/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

File: src/session/session_api.c
Comment (ruby-oujo) [inline src/session/session_api.c:635]: I don't quite understand this change, what are we using other for?

**Review by korteland — APPROVED** [review]
Wow - great work. LGTM!

**Review by ruby-oujo — APPROVED** [review]
Have one question, but changes LGTM.

File: src/session/session_api.c
Comment (tod-johnson-mongodb) [reply src/session/session_api.c:635]: other is used for additional cursors besides the cursor being opened, and owner. Most types of opens do not need other. I traced all the opens and found out with the removal of join cursors none of the statistics: cursors need other.  The call stack was like this before this change:

SESSION::open_cursor() == __session_open_cursor() -> __session_open_cursor_int(other = statjoin || dupbackup ? to_dup : NULL) -> Dispatch on uri prefix, if statistics: -> __wt_curstat_open(other) -> __wt_curstat_init(other), __wt_cursor_init().

After this change it changed to:
SESSION::open_cursor() == __session_open_cursor() -> __session_open_cursor_int(other = dupbackup ? to_dup : NULL) -> Dispatch on uri prefix, if statistics: -> __wt_curstat_open() -> __wt_curstat_init(other), __wt_cursor_init().

__session_open_cursor_int() is no longer called with other != NULL for a statistics: cursor, only for duplicating a backup cursor. Thus there is no longer a need to pass other to __wt_curstat_open() since it is always NULL.

Comment (tod-johnson-mongodb) [reply src/session/session_api.c:635]: One more thing. join: cursors used other as the cursor that is being joined to in complex joins, I think.

Comment (ruby-oujo) [reply src/session/session_api.c:635]: Thanks for the explanation!

**Review by ddanderson — COMMENTED** [review]
@tod-johnson-mongodb this is nice work.
But aren't there references to join in examples/c? src/docs?  test/suite, test/csuite?  I'm surprised the PR testing didn't catch some of these.

**Comment by tod-johnson-mongodb** [comment]
> @tod-johnson-mongodb this is nice work. But aren't there references to join in examples/c? src/docs? test/suite, test/csuite? I'm surprised the PR testing didn't catch some of these.

No, there are not references. They were already removed by https://jira.mongodb.org/browse/WT-12486 in order to pass Evergreen. The remaining documentation of SESSION::join and join cursors will be removed in https://jira.mongodb.org/browse/WT-12488.

**Pull request merged**
